### PR TITLE
Fix validation issue with hidden submission's sections

### LINKED ISSUE #2169: Submission with a validation error on an hidden step can't be submitted
**Describe the bug**
Creating a new submission, where the form definition includes a mandatory metadata within a section that is hidden for the current scope (submission/workflow), leads to the inability to deposit the submission. This is because on rest side the validation process is done for all the sections, included the hidden ones for the current scope, and it gets an error that can't be fixed by the user because the section is not accessible.   

**To Reproduce**
1. Configure the rest server as following:
  - in the `submission-form.xml` change the `traditionalpagetwo` form in order to have one metadata as mandatory (e.g `dc.description.abstract`)
```
                <field>
                    <dc-schema>dc</dc-schema>
                    <dc-element>description</dc-element>
                    <dc-qualifier>abstract</dc-qualifier>
                    <repeatable>false</repeatable>
                    <label>Abstract</label>
                    <input-type>textarea</input-type>
                    <hint>Enter the abstract of the item.</hint>
                    <required>required</required>
                </field>
```
  - in the item-submission.xml make the `traditionalpagetwo` as visible only for the workflow step
```
<scope visibility="hidden" visibilityOutside="hidden">workflow</scope>
```
2. start the application
3. with a submitter create a new submission using an entity type which form definition use the `traditionalpagetwo` section, (e.g `Publication`)
4. Fill out all the mandatory fields and try to submit the submission
5. it should be not possible even if all the mandatory have been filled out, and a warning notice should be shown instead
Schermata da 2023-03-28 11-00-08

**Expected behavior**
The validation process should take into account only the sections that are visible for the current scope

**Related work**
Link to any related tickets or PRs here.

### PR DESCRIPTION
**Pull request by atarix83**
## References
* Fixes https://github.com/DSpace/dspace-angular/issues/2169
* Fronted PR: https://github.com/DSpace/dspace-angular/pull/2243

## Description
With this PR the submission's sections that are hidden for the specific submission scope (workspace/workflow) are no longer returned into the submission process so they are not taken into account during the validation process.

## Instructions for Reviewers
Follow instructions present in the https://github.com/DSpace/dspace-angular/issues/2169 and check that is fixed

List of changes in this PR:
* The submission's sections that are hidden for the current scope are not returned in the submission object
* The submission's sections are validated only if visible for the scope

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [x] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

**Review by tdonohue — CHANGES_REQUESTED** [review]
@atarix83 : Thanks! This looks good overall and I've verified that it fixes the originally reported bug.  So, I'm nearly a +1.

However, I have a few minor  requests for missing Javadoc and also a (possibly) missing integration test.  Beyond that though, this looks good and works.

File: dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java [hunk 1]
```diff
@@ -8566,4 +8566,33 @@ public void patchBySupervisorTest() throws Exception {
...
+
+        context.restoreAuthSystemState();
+
+        getClient(getAuthToken(admin.getEmail(), password))
+            .perform(get("/api/submission/workspaceitems/" + workspaceItem.getID()))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.sections.test-outside-workflow-hidden").doesNotExist())
+            .andExpect(jsonPath("$.sections.test-outside-submission-hidden").exists())
+            .andExpect(jsonPath("$.sections.test-never-hidden").exists())
+            .andExpect(jsonPath("$.sections.test-always-hidden").doesNotExist());
```
[the remark this answers is not in the document]
Comment (atarix83) [reply dspace-server-webapp/src/test/java/org/dspace/app/rest/WorkspaceItemRestRepositoryIT.java:8597]: done

File: dspace-api/src/main/java/org/dspace/app/util/SubmissionStepConfig.java [hunk 2]
```diff
@@ -173,6 +176,32 @@ public String getVisibilityOutside() {
         return visibilityOutside;
     }
 
+    public boolean isHiddenForInProgressSubmission(InProgressSubmission obj) {
```
[the remark this answers is not in the document]
Comment (atarix83) [reply dspace-api/src/main/java/org/dspace/app/util/SubmissionStepConfig.java:185]: done

**Review by tdonohue — APPROVED** [review]
👍 Thanks @atarix83 !  This looks correct to me now and works.  As soon as it completes it's automated tests / CI process, I'll merge it.

**Pull request merged**